Updated Is "Migration Right for you?" page and some dependencies (#10158)
opensearch-project/documentation-website: _migration-assistant/overview/is-migration-assistant-right-for-you.md

Before using Migration Assistant, review the following assumptions and limitations. They are grouped by scope to clarify which apply generally and which are specific to components.

### General

- If deploying on AWS, the identity used to deploy Migration Assistant must have permission to install all required resources. For a full list of services, see [AWS Services in this Solution](https://docs.aws.amazon.com/solutions/latest/migration-assistant-for-amazon-opensearch-service/architecture-details.html#aws-services-in-this-solution).
- The target cluster must be deployed and reachable by Migration Assistant components, including the Migration Console, Reindex-from-Snapshot, and Traffic Replayer, depending on which features are used.
- The `_source` field must be enabled on all indexes to be migrated. See [Source documentation]({{site.url}}{{site.baseurl}}/field-types/metadata-fields/source/) for more information.
- If deploying to AWS, ensure the `CDKToolkit` stack exists and is in the `CREATE_COMPLETE` state. For setup instructions, see the [CDK Toolkit documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html).

> ### Reindex-from-Snapshot

Reviewer: Just a suggestion to include under RFS assumptions : 
- User should perform a successful `console metadata migrate` before performing `console backfill start` or else RFS will create new indices with default mappings on the target cluster.

Author: This order of execution will be important for the next documentation refactor which will focus on the execution steps.

- The source cluster must have the Amazon S3 plugin installed.
- Snapshots must include global cluster state (`include_global_state` is `true`).

> - Shards up to 80 GiB are supported by default. Larger shards can be configured, except in AWS GovCloud, where 80 GiB is the maximum supported size.

Reviewer: We should also answer the 'how to configure ?' question over here, by pointing them to the [this CDK step](https://docs.opensearch.org/docs/latest/migration-assistant/deploying-migration-assistant/getting-started-data-migration/#step-4-configure-and-deploy-rfs-20-minutes) during deployment.

Author: This will be included in execution. I will also move the bullet point ` Snapshots must include global cluster state (`include_global_state` is `true`).` into execution. At this point, I'd like the user to understand what limitations exist that would prevent them from using migration assistant.

### General

- If deploying on AWS, the identity used to deploy Migration Assistant must have permission to install all required resources. For a full list of services, see [AWS Services in this Solution](https://docs.aws.amazon.com/solutions/latest/migration-assistant-for-amazon-opensearch-service/architecture-details.html#aws-services-in-this-solution).
- The target cluster must be deployed and reachable by Migration Assistant components, including the Migration Console, Reindex-from-Snapshot, and Traffic Replayer, depending on which features are used.
- The `_source` field must be enabled on all indexes to be migrated. See [Source documentation]({{site.url}}{{site.baseurl}}/field-types/metadata-fields/source/) for more information.
- If deploying to AWS, ensure the `CDKToolkit` stack exists and is in the `CREATE_COMPLETE` state. For setup instructions, see the [CDK Toolkit documentation](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html).

> - If deploying Migration Assistant into an existing AWS VPC, you must configure VPC interface endpoints and IAM permissions for the following AWS services:

Reviewer 2: Where would we put that the boostrap box also needs internet access to github and docker?

Author: Updated.

Author: > There could be more additional clarity here in how we word this. In cases where VPC has network connectivity, none of these are needed, since the calls can traverse the internet to the correct AWS Service. Perhaps we can word this as "Migration Assistant may use the following AWS services and needs network access configured to them" So that network access could be provided through a NAT gateway or VPC Endpoint

Updated.

  - **Amazon CloudWatch Logs** – for log ingestion from ECS tasks.

>   - **Amazon CloudWatch** – for metrics published during migration

Reviewer 2: Let's add S3 onto this list

>   - **Amazon Elastic Container Registry (ECR)** – for pulling container images.

Reviewer 2: is this enough detail for customers? there are two ECR endpoints
`com.amazonaws.region.ecr.dkr` and `com.amazonaws.region.ecr.api`, we need the former to pull images not the later

https://docs.aws.amazon.com/AmazonECR/latest/userguide/vpc-endpoints.html#ecr-setting-up-vpc-create

  - **Amazon Elastic Container Service (ECS)** – for task orchestration and migration.

>   - **Elastic Load Balancing (ELB)** – for routing traffic to Capture Proxy.

Reviewer 2: This was added to be able to configure the ALB routing from the migration console, it's not needed for routing traffic

## Migration Assistant assumptions and limitations

Before using Migration Assistant, review the following assumptions and limitations.

### Networking and Environment

Migration Assistant requires network connectivity to AWS services and outbound internet access to build and deploy. Requirements differ based on whether you're deploying into a new or existing VPC.

#### Source and Target Connectivity

To meet connectivity requirements, ensure the following:

- You must establish connectivity between:
  - The source cluster and/or Amazon S3 (for snapshots, which may only require updating the bucket policy) and Migration Assistant
  - The target cluster and Migration Assistant
- If the source or target resides in a private VPC without internet access, use one of the following to connect:
  - VPC endpoints
  - VPC peering
  - AWS Transit Gateway

#### Deploying into a New VPC

When deploying into a new VPC, consider the following:

- Migration Assistant provisions a new VPC with required components (for example, NAT gateway, subnets).
- You must establish network access from this VPC to both the source and target clusters.

> #### Deploying into an Existing VPC

Reviewer 3: ```suggestion
#### Deploying into an existing VPC
```

- Do you need to validate a new OpenSearch cluster before switching over, with rollback capabilities?
- Is your environment self-managed or running on Amazon OpenSearch Service?
- Are you looking for tooling to migrate index settings and other metadata?
- Do you need to reconfigure your target cluster—for example, by changing the sharding strategy and reindexing?
- Are you migrating across regions, from on-premises, or from another cloud provider?
- Do you need a high-performance backfill solution that can reliably reindex documents—with support for pause, resume, or checkpoint recovery?

If you answered "yes" to most of these questions, Migration Assistant is likely the right solution for your migration.

## Migration Assistant assumptions and limitations

Before using Migration Assistant, review the following assumptions and limitations.

### Networking and Environment

Migration Assistant requires network connectivity to AWS services and outbound internet access to build and deploy. Requirements differ based on whether you're deploying into a new or existing VPC.

#### Source and Target Connectivity

- You must establish connectivity between:

>   - The source cluster and/or Amazon S3 (for snapshots) and Migration Assistant

Author: updated

  - The target cluster and Migration Assistant
- If the source or target resides in a private VPC without internet access, use one of the following to connect:
  - VPC endpoints
  - VPC peering
  - AWS Transit Gateway

#### Deploying into a New VPC

- Migration Assistant provisions a new VPC with required components (e.g., NAT gateway, subnets).

> - You must establish network access from this VPC to both the source (or S3) and target clusters.

Reviewer 2: nit: S3 is included by default in this "new VPC" flow

    - **Amazon Application Load Balancer (ALB)** – *(`Capture-and-Replay` only)* - Used to optionally reroute client traffic from the source to the target during migration.
    - **Amazon CloudWatch** – Publishes migration metrics.
    - **Amazon CloudWatch Logs** – Ingests ECS task logs.
    - **Amazon EC2** – Used to bootstrap Migration Assistant. When deploying using CloudFormation, the bootstrap EC2 instance requires outbound internet access (using NAT gateway or IGW) to download the latest release from GitHub.
    - **Amazon Elastic Block Store (Amazon EBS)** – Provides temporary disk storage.
    - **Amazon Elastic Container Registry (Amazon ECR)** – Pulls container images.
    - **Amazon Elastic Container Service (Amazon ECS)** – Orchestrates container workloads.
    - **Amazon Elastic File System (Amazon EFS)** – Stores persistent logs.
    - **Amazon Managed Streaming for Apache Kafka** – *(`Capture-and-Replay` only)* - Used as a durable storage to capture and replay HTTP live traffic.
    - **Amazon S3** – Stores and retrieves snapshots and artifacts.

>     - **AWS Elastic Load Balancing (ELB)** – *(Capture-and-Replay only)* – Used by the Migration Console to connect to the Application Load Balancer (ALB).

Reviewer 4: @AndreKurait @sumobrian Global editorial note: Please use "Elastic Load Balancing" (not ELB) and "Application Load Balancer" (without ALB) in all instances. Related resources include Network Load Balancer, Gateway Load Balancer, and Classic Load Balancer. Use just "load balancer" when referencing the common noun.

>     - **AWS Elastic Load Balancing (ELB)** – *(Capture-and-Replay only)* – Used by the Migration Console to connect to the Application Load Balancer (ALB).

Author: I updated references and noted for future changes. I left the acronym in parenthesis since often users are more familiar with the acronym.

>     - **AWS Secrets Manager** – Stores credentials securely.

Reviewer 5: nit: We could say only needed when using basic auth for source or target cluster

### Backfill migration

Use [backfill migration]({{site.url}}{{site.baseurl}}/migration-assistant/migration-phases/backfill/) if:

- You need to move historical data without disrupting live traffic.
- You want to backfill indexes from a specific point in time without impacting the source cluster.
- You want to verify historical data in the target cluster before switching over.
- You want to backfill using an existing or incremental snapshot.
- You need the fastest backfill option that includes reindexing.
- You want the ability to pause and resume migration.

### RFS

Use [RFS]({{site.url}}{{site.baseurl}}/migration-assistant/deploying-migration-assistant/getting-started-data-migration/) if:

- You already use OpenSearch snapshots for backups.
- You need to migrate documents at scale in parallel, such as with Amazon Elastic Container Service (Amazon ECS).
- You require a data migration path as part of a zero-downtime migration.
- Your AWS Region supports RFS and your shard sizes are within supported limits.

### Combination of all three

Use a combination of all three migration types if:

- You're performing a complex, multi-version migration.

> - You require zero downtime and full validation of the target environment.

Author: This is coming in a subsequent PR. https://github.com/opensearch-project/documentation-website/pull/10169

#### Source and Target Connectivity

- You must establish connectivity between:

>   - The source cluster and/or Amazon S3 (for snapshots, which may only require updating the bucket policy) and Migration Assistant

Reviewer 5: nit: My preference of wording
```suggestion
  - The source cluster and/or Amazon S3 (if using [Bring your own snapshot]({{site.url}}{{site.baseurl}}/migration-assistant/deploying-migration-assistant/configuration-options/#bring-your-own-snapshot)) and Migration Assistant
```
